python/taichi/lang/__init__.py

```diff
@@ -565,6 +565,9 @@ def block_local(*args):
     Args:
         *args (List[Field]): A list of sparse Taichi fields.
     """
+    if impl.current_cfg().opt_level == 0:
+        warning("""opt_level = 1 is enforced to enable bls analysis.""")
```

Comment:
for consistency with other code, shall we use `ti.warn`? ;)

Reply:
https://github.com/taichi-dev/taichi/pull/3434/commits/9ccdd16813af75332c2fe738a4c5763ecab86476